tests/AppBundle/Listener/ListenerTest.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace Tests\AppBundle\Listener;
+
+use AppBundle\Listener\AccessListener;
+use AppBundle\Listener\ResponseListener;
+use AppBundle\Security\RateLimiter;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+use Symfony\Component\HttpKernel\Controller\ControllerResolver;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernel;
+
+class ListenerTest extends KernelTestCase
+{
+    /**
+     * @var \Doctrine\ORM\EntityManager
+     */
+    private $em;
+
+    protected function setUp()
+    {
+        self::bootKernel();
+
+        $this->em = static::$kernel->getContainer()
+            ->get('doctrine')
+            ->getManager();
+    }
+
+    private function getResponseEvent()
+    {
+        $request = Request::create(
+            '/videos',
+            'GET'
+        );
+        $dispatcher = new EventDispatcher();
+
+        $controllerResolver = new ControllerResolver();
+        $argumentResolver = new ArgumentResolver();
+
+        $kernel = new HttpKernel($dispatcher, $controllerResolver, new RequestStack(), $argumentResolver);
+
+        $event = new GetResponseEvent($kernel, $request, 1);
+
+        return $event;
+    }
+
+    private function getFilterResponseEvent()
+    {
+        $request = Request::create(
+            '/videos',
+            'GET'
+        );
+        $dispatcher = new EventDispatcher();
+
+        $controllerResolver = new ControllerResolver();
+        $argumentResolver = new ArgumentResolver();
+
+        $kernel = new HttpKernel($dispatcher, $controllerResolver, new RequestStack(), $argumentResolver);
+
+        $response = new Response();
+        $response->setContent(json_encode(
+            ['data' => '123']
+        ));
+
+        $event = new FilterResponseEvent($kernel, $request, 1, $response);
+
+        return $event;
+    }
+
+    public function testRequestPassed()
+    {
+        $rateLimiter = new RateLimiter($this->em);
```

Comment:
You should mock the `RateLimiter` here because you are testing only the Listener. This will avoid false positives if the RateLimiter is changed.